File: GomokuServer/src/GomokuServer.Application/Games/Responses/GetAvailableGamesResponse.cs
namespace GomokuServer.Application.Games.Responses;

public record GetAvailableGamesResponse(string GameId)
{
	[Required]
	public string GameId { get; } = GameId;

	public ProfileDto? Opponent { get; init; }

	public TimeControlDto? TimeControl { get; init; }
}

File: GomokuClient/packages/gomoku-api/client/models/GetAvailableGamesResponse.ts
import type { ProfileDto } from "./ProfileDto";
import type { TimeControlDto } from "./TimeControlDto";

export type GetAvailableGamesResponse = {
  /**
   * @type string
   */
  gameId: string;
  /**
   * @type object | undefined
   */
  opponent?: ProfileDto;
  /**
   * @type object | undefined
   */
  timeControl?: TimeControlDto;
};
